src/ut/utils_test.cpp

```diff
@@ -172,3 +172,103 @@ TEST_F(UtilsTest, RemoveBracketsFromBareIPv6Address)
 {
   EXPECT_EQ("::1", remove_brackets_from_ip("::1"));
 }
+
+TEST_F(UtilsTest, IsAWildcard)
+{
+  EXPECT_TRUE(is_wildcard_uri("sip:!.!*!@domain"));
+}
+
+TEST_F(UtilsTest, IsNotAWildcard)
+{
+  EXPECT_FALSE(is_wildcard_uri("sip:!.*@domain"));
```

Comment:
Similar to above, I think we should have more tests here...
* Longer URIs with `!` in different places
* No `!`
* tel URIs?
* `!` in the domain part (is that allowed?)